### PR TITLE
turnToHeading()

### PR DESCRIPTION
**Pull request by nic-cheng**
#### Summary
Implements turn to heading function, renamed old function to `turnToPoint()`.

#### Motivation
Turning to a heading instead of a point is also useful.

#### Test Plan
Plan: run on a robot the same way turnToPoint and other motions would be tested.

#### Additional Notes
Runs with slightly modified logic from the old `turnTo()` so there should be no issues (unless I screwed something up).
Tested by @edwart2cAKE.

## Download the template for this pull request: 

> [!NOTE]  
> This is auto generated from [`Add Template to Pull Request`](https://github.com/LemLib/LemLib/actions/runs/7696942468)
- via manual download: [LemLib@0.5.0-rc.6+015114.zip](https://nightly.link/LemLib/LemLib/actions/artifacts/1202490140.zip)
- via PROS Integrated Terminal: 
 ```
curl -o LemLib@0.5.0-rc.6+015114.zip https://nightly.link/LemLib/LemLib/actions/artifacts/1202490140.zip;
pros c fetch LemLib@0.5.0-rc.6+015114.zip;
pros c apply LemLib@0.5.0-rc.6+015114;
rm LemLib@0.5.0-rc.6+015114.zip;
```

### CODE REVIEW

File: include/lemlib/chassis/chassis.hpp [hunk 1]
```diff
@@ -236,7 +236,22 @@ class Chassis {
...
+         *
+         * @param targetTheta robot target heading
+         * @param timeout longest time the robot can spend moving
+         * @param radians whether the heading is in radians or degrees. false by default
+         * @param forwards whether the robot should turn to face the heading with the front of the robot. true by
+         * default
+         * @param maxSpeed the maximum speed the robot can turn at. Default is 127
+         * @param async whether the function should be run asynchronously. true by default
+         */
+        void turnToHeading(float targetTheta, int timeout, bool radians = false, bool forwards = true,
```
Comment (liam-teale) [inline include/lemlib/chassis/chassis.hpp]: there should not be an option for radians. It's highly unlikely a user will use radians, and it makes the function call more complex

File: src/lemlib/chassis/chassis.cpp [hunk 2]
```diff
@@ -279,6 +279,78 @@ void lemlib::Chassis::turnTo(float x, float y, int timeout, bool forwards, float
...
+        distTravelled = fabs(angleError(pose.theta, startTheta));
+
+        // calculate deltaTheta
+        deltaTheta = angleError(targetTheta, pose.theta, false);
+
+        // calculate the speed
+        motorPower = angularPID.update(deltaTheta);
+        angularLargeExit.update(deltaTheta);
+        angularSmallExit.update(deltaTheta);
+
```
Comment (liam-teale) [inline src/lemlib/chassis/chassis.cpp:540]: The output should be slewed unless its within 20 degrees of the target

File: src/main.cpp [hunk 3]
```diff
@@ -131,9 +132,13 @@ void autonomous() {
     // cancel the movement after it has travelled 10 inches
     chassis.waitUntil(10);
     chassis.cancelMotion();
+    // example movement: Turn to face a heading of 90º. Timeout set to 1500
+    chassis.turnToHeading(1000, 1500);
+    // example movement: Turn to face a heading of pi radians. Timeout set to 1500
+    chassis.turnToHeading(M_PI, 1500, true);
```
Comment (liam-teale) [inline src/main.cpp]: remember to remove this when you remove the radians option

**Review by liam-teale — CHANGES_REQUESTED** [review]

**Review by liam-teale — APPROVED** [review]
LGTM, will merge upon testing

**Comment by liam-teale** [comment]
Has this been tested yet?

File: .DS_Store
Comment (liam-teale) [inline .DS_Store:1]: This should not be here. Remove it, and I'll also add it to the gitignore

File: include/lemlib/api.hpp [hunk 4]
```diff
@@ -10,10 +10,11 @@
  */
 #pragma once
 
-#include "lemlib/util.hpp"
 #include "lemlib/pid.hpp"
 #include "lemlib/pose.hpp"
-#include "lemlib/chassis/trackingWheel.hpp"
+#include "lemlib/util.hpp"
+
```
Comment (liam-teale) [inline include/lemlib/api.hpp]: why the newlines in this file?

**Review by liam-teale — CHANGES_REQUESTED** [review]

**Comment by nic-cheng** [comment]
NOTE: hasn't been fully tested yet, @edwart2cAKE is testing it.

**Comment by nic-cheng** [comment]
Update: passed testing by @edwart2cAKE
image

**Comment by Rocky14683** [comment]
LGTM

**Pull request merged**